### DISCUSSION TITLE
Challenges with Time Discretization

### DISCUSSION THREAD

**Discussion by Maguixx** [pybamm-team/liionpack#301] [Q&A]
Hello everyone,
 I've been working on battery simulation using liionpack, integrated with PyBaMM, to conduct thermal simulations of Li-ion battery packs. I'm trying to replicate results from a specific experiment but have encountered difficulties with the time discretization aspect.
 
This is the experiment I am using:
 
experiment = pybamm.Experiment(
    ["Discharge at 240A for 14 seconds", "Charge at 40A for 26 seconds"],
    period="2 seconds",
)
The challenge arises when I attempt to adjust the period parameter to a smaller timescale, such as 0.01 seconds. According to the liionpack documentation, the time should be an integer multiple of the period, which restricts the level of granularity I can achieve in the simulations.
 
My primary concern is the potential impact of these restrictions on the accuracy and the resolution of the simulation results, especially when comparing them to detailed experimental data or other high-resolution simulation outputs.
 
Could anyone share insights or workarounds on how to handle finer time discretization in liionpack simulations? Is there a way to override the integer-multiple requirement for the period, or perhaps another method to achieve higher resolution in temporal terms without compromising the stability of the simulation?

Thank you for your assistance.

**Comment by TomTranter** [comment]
I think this should be fixed in the latest release but let me know
